From f12ea3ecc41328f090f03453d785ad79f9e46b34 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Fri, 24 Apr 2026 14:18:04 +0900
Subject: [PATCH 04/15] DEV: conf: Add yaml files for kas

Usage:
  KAS_WORK_DIR=kas-work kas build conf/kas/minimal.yaml
  KAS_WORK_DIR=kas-work kas build conf/kas/weston.yaml

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 .gitignore                                |  1 +
 conf/kas/common.yaml                      | 98 +++++++++++++++++++++++
 conf/kas/minimal.yaml                     | 11 +++
 conf/kas/option/enable-sbom.yaml          | 30 +++++++
 conf/kas/option/enable-sdk-sources.yaml   |  8 ++
 conf/kas/option/enable-sstate-mirror.yaml | 10 +++
 conf/kas/option/kernel-6.12.yaml          |  8 ++
 conf/kas/option/kernel-6.18.yaml          |  8 ++
 conf/kas/option/kernel-dummy.yaml         |  7 ++
 conf/kas/option/rm-work.yaml              |  7 ++
 conf/kas/weston.yaml                      | 19 +++++
 11 files changed, 207 insertions(+)
 create mode 100644 conf/kas/common.yaml
 create mode 100644 conf/kas/minimal.yaml
 create mode 100644 conf/kas/option/enable-sbom.yaml
 create mode 100644 conf/kas/option/enable-sdk-sources.yaml
 create mode 100644 conf/kas/option/enable-sstate-mirror.yaml
 create mode 100644 conf/kas/option/kernel-6.12.yaml
 create mode 100644 conf/kas/option/kernel-6.18.yaml
 create mode 100644 conf/kas/option/kernel-dummy.yaml
 create mode 100644 conf/kas/option/rm-work.yaml
 create mode 100644 conf/kas/weston.yaml

diff --git a/.gitignore b/.gitignore
index fbaacbd..673a0e4 100644
--- a/.gitignore
+++ b/.gitignore
@@ -2,3 +2,4 @@ build
 firmware
 /.venv
 /scripts/*
+/kas-work
diff --git a/conf/kas/common.yaml b/conf/kas/common.yaml
new file mode 100644
index 0000000..f13c7d9
--- /dev/null
+++ b/conf/kas/common.yaml
@@ -0,0 +1,98 @@
+# Every file needs to contain a header, that provides kas with information
+# about the context of this file.
+header:
+  # The `version` entry in the header describes for which configuration
+  # format version this file was created for. It is used by kas to figure
+  # out if it is compatible with this file. The version is an integer that
+  # is increased on every format change.
+  version: 1
+
+env:
+  KAS_WORK_DIR: "./kas-work"
+
+local_conf_header:
+  meta-custom: |
+    CONF_VERSION = "2"
+    MACHINE ??= "sparrow-hawk"
+    DISTRO ?= "poky"
+    PACKAGE_CLASSES ?= "package_deb"
+    EXTRA_IMAGE_FEATURES ?= "allow-empty-password empty-root-password allow-root-login"
+    USER_CLASSES ?= "buildstats"
+    PATCHRESOLVE = "noop"
+    BB_DISKMON_DIRS ??= "\
+      STOPTASKS,${TMPDIR},1G,100K \
+      STOPTASKS,${DL_DIR},1G,100K \
+      STOPTASKS,${SSTATE_DIR},1G,100K \
+      STOPTASKS,/tmp,100M,100K \
+      HALT,${TMPDIR},100M,1K \
+      HALT,${DL_DIR},100M,1K \
+      HALT,${SSTATE_DIR},100M,1K \
+      HALT,/tmp,10M,1K"
+    PACKAGECONFIG:append:pn-qemu-system-native = " sdl"
+    INIT_MANAGER = "systemd"
+    ROOT_HOME = '/root'
+
+    # add the static lib to SDK toolchain
+    SDKIMAGE_FEATURES:append = " staticdev-pkgs"
+
+    DISTRO_FEATURES:remove = "ptest"
+    LICENSE_FLAGS_ACCEPTED = "non-commercial"
+
+    # Install license manifest and license texts into rootfs
+    COPY_LIC_MANIFEST = "1"
+    COPY_LIC_DIRS = "1"
+
+    # Reproducible Builds
+    export SOURCE_DATE_EPOCH = "${@time.strftime('%s', time.localtime())}"
+    REPRODUCIBLE_TIMESTAMP_ROOTFS = "${@time.strftime('%s', time.localtime())}"
+
+    PREFERRED_VERSION_linux-renesas = "6.18%"
+    LINUXLIBCVERSION = "6.18%"
+    # linux-libc-headers is depended on Yocto version
+    PREFERRED_VERSION_linux-libc-headers = "6.6"
+    PREFERRED_VERSION_nativesdk-linux-libc-headers = "6.6"
+
+    # Disable create-spdx
+    INHERIT:remove = "create-spdx"
+
+    # Reduce fetch size
+    BB_GIT_SHALLOW = "1"
+    BB_GENERATE_MIRROR_TARBALLS = "1"
+    BB_GIT_SHALLOW_DEPTH = "1"
+
+
+# The machine as it is written into the `local.conf` of bitbake.
+machine: sparrow-hawk
+distro: poky
+
+# The distro name as it is written into the `local.conf` of bitbake.
+distro: poky
+
+repos:
+  # This entry includes the repository where the config file is located
+  # to the bblayers.conf:
+  meta-custom:
+  # Here we include a list of layers from the poky repository to the
+  # bblayers.conf:
+  poky:
+    path: layers/poky
+    url: "https://git.yoctoproject.org/poky"
+    branch: scarthgap
+    #commit:
+    layers:
+      meta:
+      meta-poky:
+      meta-yocto-bsp:
+
+  meta-openembedded:
+    path: layers/meta-openembedded
+    url: "https://git.openembedded.org/meta-openembedded"
+    branch: scarthgap
+    #commit:
+    layers:
+      meta-oe:
+      meta-python:
+      meta-networking:
+      meta-multimedia:
+
+
diff --git a/conf/kas/minimal.yaml b/conf/kas/minimal.yaml
new file mode 100644
index 0000000..5cff57d
--- /dev/null
+++ b/conf/kas/minimal.yaml
@@ -0,0 +1,11 @@
+header:
+  version: 1
+  includes:
+    - conf/kas/common.yaml
+
+local_conf_header:
+  minimal: |
+    BBMASK = "kernel-module-gles|gles-user-module|wayland-wsegl|wayland-kms|libgbm"
+
+target: core-image-minimal
+
diff --git a/conf/kas/option/enable-sbom.yaml b/conf/kas/option/enable-sbom.yaml
new file mode 100644
index 0000000..33f653a
--- /dev/null
+++ b/conf/kas/option/enable-sbom.yaml
@@ -0,0 +1,30 @@
+
+header:
+  version: 1
+
+local_conf_header:
+  meta-custom-sbom: |
+    # required. enable to generate spdx files.
+    INHERIT:append = " create-spdx"
+
+    # optional. if "1", output spdx files will be formatted.
+    SPDX_PRETTY = "1"
+
+    # optional. if "1", output spdx files includes https://spdx.github.io/spdx-spec/v2.3/file-information/
+    SPDX_INCLUDE_SOURCES = "1"
+
+    # optional. if "1", bitbake will create output binary archive for each package.
+    SPDX_ARCHIVE_PACKAGED = "1"
+
+    # required. archive the source of GPL/LGPL/AGPL packages
+    # into tmp/deploy/sources.
+    INHERIT:append = " archiver"
+
+    # optional. archive the original source, patches and recipe.
+    ARCHIVER_MODE[src] = "original"
+    ARCHIVER_MODE[diff] = "1"
+    ARCHIVER_MODE[recipe] = "1"
+
+    # optional. archive the source only of the recipes included in the output.
+    # SDK build uses enable-sdk-sources.yaml to add nativesdk and cross-canadian.
+    COPYLEFT_RECIPE_TYPES ?= "target"
diff --git a/conf/kas/option/enable-sdk-sources.yaml b/conf/kas/option/enable-sdk-sources.yaml
new file mode 100644
index 0000000..7a5c729
--- /dev/null
+++ b/conf/kas/option/enable-sdk-sources.yaml
@@ -0,0 +1,8 @@
+header:
+  version: 1
+
+local_conf_header:
+  meta-custom-sdk-sources: |
+    # The SDK installer includes nativesdk and cross-canadian (gcc, binutils)
+    # recipes, so archive the source of them too.
+    COPYLEFT_RECIPE_TYPES = "target nativesdk cross-canadian"
diff --git a/conf/kas/option/enable-sstate-mirror.yaml b/conf/kas/option/enable-sstate-mirror.yaml
new file mode 100644
index 0000000..0ad7ec3
--- /dev/null
+++ b/conf/kas/option/enable-sstate-mirror.yaml
@@ -0,0 +1,10 @@
+header:
+  version: 1
+
+local_conf_header:
+  meta-custom-sstate-mirror: |
+    BB_HASHSERVE_UPSTREAM = "wss://hashserv.yoctoproject.org/ws"
+    SSTATE_MIRRORS ?=  "file://.* http://cdn.jsdelivr.net/yocto/sstate/all/PATH;downloadfilename=PATH"
+    BB_HASHSERVE = "auto"
+    BB_SIGNATURE_HANDLER = "OEEquivHash"
+
diff --git a/conf/kas/option/kernel-6.12.yaml b/conf/kas/option/kernel-6.12.yaml
new file mode 100644
index 0000000..ba84eb3
--- /dev/null
+++ b/conf/kas/option/kernel-6.12.yaml
@@ -0,0 +1,8 @@
+header:
+  version: 1
+
+local_conf_header:
+  meta-custom-kenrel-612: |
+    PREFERRED_VERSION_linux-renesas = "6.12"
+    LINUXLIBCVERSION = "6.12"
+
diff --git a/conf/kas/option/kernel-6.18.yaml b/conf/kas/option/kernel-6.18.yaml
new file mode 100644
index 0000000..f0044d9
--- /dev/null
+++ b/conf/kas/option/kernel-6.18.yaml
@@ -0,0 +1,8 @@
+header:
+  version: 1
+
+local_conf_header:
+  meta-custom-kernel-618: |
+    PREFERRED_VERSION_linux-renesas = "6.18"
+    LINUXLIBCVERSION = "6.18"
+
diff --git a/conf/kas/option/kernel-dummy.yaml b/conf/kas/option/kernel-dummy.yaml
new file mode 100644
index 0000000..7f71c43
--- /dev/null
+++ b/conf/kas/option/kernel-dummy.yaml
@@ -0,0 +1,7 @@
+header:
+  version: 1
+
+local_conf_header:
+  meta-custom-kernel-dummy: |
+    PREFERRED_PROVIDER_virtual/kernel = "linux-dummy"
+
diff --git a/conf/kas/option/rm-work.yaml b/conf/kas/option/rm-work.yaml
new file mode 100644
index 0000000..0e43534
--- /dev/null
+++ b/conf/kas/option/rm-work.yaml
@@ -0,0 +1,7 @@
+header:
+  version: 1
+
+local_conf_header:
+  meta-custom-rm-work: |
+    INHERIT:append = " rm_work"
+
diff --git a/conf/kas/weston.yaml b/conf/kas/weston.yaml
new file mode 100644
index 0000000..a62b772
--- /dev/null
+++ b/conf/kas/weston.yaml
@@ -0,0 +1,19 @@
+header:
+  version: 1
+  includes:
+    - conf/kas/common.yaml
+
+local_conf_header:
+  weston: |
+    # Enable Wayland/Weston
+    DISTRO_FEATURES_NATIVESDK:append = " wayland"
+    DISTRO_FEATURES:append = " pam"
+
+    # To build gstreamer plugins
+    LICENSE_FLAGS_ACCEPTED = "commercial"
+
+    # Remove ptest to reduce the build time
+    # Remove x11 to avoid error after removing mesa
+    DISTRO_FEATURES:remove = "ptest x11"
+
+target: core-image-weston
